### PR TITLE
RoutingNG: Fix infinite loop caused by history state not being cleaned up

### PR DESCRIPTION
**Pull request by dprokop**
Ref https://github.com/grafana/grafana/issues/31873

This is a quick fix for the issue, the history.state solution for force-reloading routes is for me to rethink next week

### CODE REVIEW

**Review by torkelo — COMMENTED** [review]
I was thinking another option to a Boolean is a counter , a counter we do not have to reset , only check that it changed

**Comment by dprokop** [comment]
>I was thinking another option to a Boolean is a counter , a counter we do not have to reset , only check that it changed

@torkelo tired that, seems to work OK. Only applied to Dashboard page as this is the only page we use it.

File: packages/grafana-runtime/src/services/LocationService.ts [hunk 1]
```diff
@@ -109,9 +110,10 @@ export class HistoryWrapper implements LocationService {
   }
 
   reload() {
+    const prevState = (this.history.location.state as any)?.forceRouteReload;
     this.history.replace({
       ...this.history.location,
-      state: { forceRouteReload: true },
+      state: { forceRouteReload: prevState ? prevState + 1 : 1 },
```
Comment (torkelo) [inline packages/grafana-runtime/src/services/LocationService.ts:116]: I would name the state property routeReloadCounter,

File: packages/grafana-runtime/src/services/LocationService.ts [hunk 2]
```diff
@@ -96,7 +96,8 @@ export class HistoryWrapper implements LocationService {
   }
 
   replace(location: H.Path | H.LocationDescriptor, forceRouteReload?: boolean) {
-    const state = forceRouteReload ? { forceRouteReload: true } : undefined;
+    const prevState = (this.history.location.state as any)?.forceRouteReload;
+    const state = forceRouteReload ? { forceRouteReload: prevState ? prevState + 1 : 1 } : undefined;
```
Comment (torkelo) [inline packages/grafana-runtime/src/services/LocationService.ts:100]: I think we can remove forceRouteReload?: boolean parameter from this function, and move this counter increment logic to the two places that need it (as this function already takes in the location it feels unnecessary to have a boolean option that modifies the object that can already be passed in)

**Review by torkelo — APPROVED** [review]
Looks good!  just some small nits

**Pull request merged**